samples/FunctionApp/Function3/Function3.cs

```diff
@@ -1,24 +1,27 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System.Net;
 using Microsoft.Azure.Functions.Worker;
-using Microsoft.Azure.WebJobs;
-using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Microsoft.Azure.Functions.Worker.Extensions.Http;
+using Microsoft.Azure.Functions.Worker.Extensions.Storage;
+using Microsoft.Azure.Functions.Worker.Pipeline;
 
 namespace FunctionApp
 {
     public static class Function3
     {
 
         [FunctionName("Function3")]
+        [QueueOutput("name", "functionstesting2", Connection = "AzureWebJobsStorage")]
         public static HttpResponseData Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequestData req,
-            [Queue("functionstesting2", Connection = "AzureWebJobsStorage")] OutputBinding<string> name)
+            FunctionExecutionContext context)
         {
             var response = new HttpResponseData(HttpStatusCode.OK);
             response.Body = "Success!!";
 
-            name.SetValue("some name");
+            context.OutputBindings["name"] = "some name";
```

Comment:
This might just be my opinion, but I think I prefer the "OutputBinding" implementation. A few concerns about the "QueueOutput" attribute: 1. the lack of type enforcement on the "name" property (might introduce some bugs javascript developers know too well) and 2. the lack of intellisense around context.OutputBindings (so it's not so intuitive that there would be a "name" property). I know IAsyncCollector was a bit wonky, but I wonder if it's also because the name wasn't intuitive (in how it relates to an output binding) and because it was one of many ways to do output bindings.

Reply:
As discussed offline, we are thinking of keeping this as is (without OutputBinding type). Once we write some samples using different extensions, we'll come up with scenarios where we think one approach may work better than the other, and decide on that. We will be sure to that before we release these set of changes.